api/gateway.go

```diff
@@ -41,6 +45,10 @@ func (srv *Server) gatewayAddHandler(w http.ResponseWriter, req *http.Request, p
 // gatewayRemoveHandler handles the API call to remove a peer from the gateway.
 func (srv *Server) gatewayRemoveHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
 	addr := modules.NetAddress(ps.ByName("netaddress"))
+	if err := addr.IsValid(); err != nil {
+		writeErorr(w, "invalid address: "+err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
I think it's simpler to do validation as soon as the address is created, that way validation isn't required in every method that takes a NetAddress as an argument. In this case it is true that the call will fail because the address isn't dialable, but I don't think we can always rely on this behavior unless we enforce that all exported methods that take a NetAddress also validate it.

I'm not sure which is better. What do you think?

As a side note if you pass an un-dialable NetAddress to `Connect` it will add it to it's peer list before it returns an error. This should probably be fixed regardless.